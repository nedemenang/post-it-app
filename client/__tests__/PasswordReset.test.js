import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PasswordReset from '../components/passwordReset.jsx';

describe('the password reset component', () => {
  let mountedComponent;

  const PasswordResetItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<PasswordReset />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = PasswordResetItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = PasswordResetItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should update state on email field change', () => {
    const input = PasswordResetItem().find('input').first();
    input.simulate('change', { target: {
      value: 'test@email.com'
    } });
    expect(PasswordResetItem().state().resetEmail).toBe('test@email.com');
  });

  it('should call function on submit of form', () => {
    const component = shallow(<PasswordReset />);
    const preventDefault = jest.fn();
    component.find('button').simulate('click', { preventDefault });
    expect(preventDefault).toBeCalled();
  });
});
