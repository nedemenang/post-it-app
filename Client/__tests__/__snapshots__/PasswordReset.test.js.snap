// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the password reset component should match snapshot test 1`] = `
<div
  className="login-page"
>
  <div
    className="form"
  >
    <form
      className="login-form"
    >
      <h3>
        Reset Password
      </h3>
      <input
        onChange={[Function]}
        placeholder="email"
        type="text"
      />
      <p
        className="success"
      />
      <p
        className="error"
      />
      <button
        className="button"
        onClick={[Function]}
      >
        Send Email
      </button>
      <br />
      <br />
    </form>
  </div>
</div>
`;
